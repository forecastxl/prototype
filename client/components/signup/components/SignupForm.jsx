import React, { PropTypes } from 'react'
import Checkbox from 'material-ui/Checkbox'
import RaisedButton from 'material-ui/RaisedButton'
import TextField from 'material-ui/TextField'
import { BaseErrors, TextFieldErrors } from '../../errors'

function LoginForm({ onSubmit, onTextChange, onCheckboxChange, account, errors }) {
  return (
    <div>
      {errors && errors.base && <BaseErrors errors={errors.base} />}
      <form action="/" onSubmit={onSubmit}>
        <TextField
          floatingLabelText="First Name"
          name="firstName"
          value={account.firstName}
          onChange={onTextChange}
          errorText={errors && errors.firstName && <TextFieldErrors errors={errors.firstName} />}
        /><br />
        <TextField
          floatingLabelText="Last Name"
          name="lastName"
          value={account.lastName}
          onChange={onTextChange}
          errorText={errors && errors.lastName && <TextFieldErrors errors={errors.lastName} />}
        /><br />
        <TextField
          floatingLabelText="Company Name"
          name="companyName"
          value={account.companyName}
          onChange={onTextChange}
          errorText={
            errors &&
            errors.companyName &&
            <TextFieldErrors errors={errors.companyName} />
          }
        /><br />
        <TextField
          floatingLabelText="Phone Number"
          name="phoneNumber"
          type="tel"
          value={account.phoneNumber}
          onChange={onTextChange}
          errorText={
            errors &&
            errors.phoneNumber &&
            <TextFieldErrors errors={errors.phoneNumber} />
          }
        /><br />
        <TextField
          floatingLabelText="Email"
          name="email"
          type="email"
          value={account.email}
          onChange={onTextChange}
          errorText={errors && errors.email && <TextFieldErrors errors={errors.email} />}
        /><br />
        <TextField
          floatingLabelText="Password"
          name="password"
          type="password"
          value={account.password}
          onChange={onTextChange}
          errorText={errors && errors.password && <TextFieldErrors errors={errors.password} />}
        /><br />
        <TextField
          floatingLabelText="Repeat password"
          name="passwordConfirmation"
          type="password"
          value={account.passwordConfirmation}
          onChange={onTextChange}
          errorText={
            errors &&
            errors.passwordConfirmation &&
            <TextFieldErrors errors={errors.passwordConfirmation} />
          }
        /><br />
        <Checkbox
          label="I agree to the terms and conditions"
          name="agreedToGeneralTerms"
          checked={account.agreedToGeneralTerms}
          onCheck={onCheckboxChange}
        /><br />
        <RaisedButton type="submit" label="Sign up" primary />
      </form>
    </div>
  )
}

LoginForm.propTypes = {
  errors: PropTypes.object.isRequired,
  onSubmit: PropTypes.func.isRequired,
  onTextChange: PropTypes.func.isRequired,
  onCheckboxChange: PropTypes.func.isRequired,
  account: PropTypes.shape({
    firstName: PropTypes.string.isRequired,
    lastName: PropTypes.string.isRequired,
    companyName: PropTypes.string.isRequired,
    phoneNumber: PropTypes.string.isRequired,
    email: PropTypes.string.isRequired,
    password: PropTypes.string.isRequired,
    passwordConfirmation: PropTypes.string.isRequired,
    agreedToGeneralTerms: PropTypes.bool.isRequired
  })
}

export default LoginForm
