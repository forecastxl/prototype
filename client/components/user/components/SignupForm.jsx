import React, { PropTypes } from 'react'
import Checkbox from 'material-ui/Checkbox'
import TextField from 'material-ui/TextField'
import { Spacer } from '../../spacer'
import { Alert } from '../../alert'
import { Dialog, DialogBody, DialogHeader, DialogButton } from '../../dialog'

function LoginForm({ onSubmit, onTextChange, onCheckboxChange, account, errors }) {
  return (
    <Dialog>
      <DialogHeader
        title="Create an account with ForecastXL"
        titleColor="white"
      />
      <DialogBody>
        {errors && errors.base && <Alert messages={errors.base} />}
        <form action="/" onSubmit={onSubmit}>
          <Spacer bottom>
            <TextField
              floatingLabelText="First Name"
              name="firstName"
              value={account.firstName}
              onChange={onTextChange}
              fullWidth
              errorText={
                errors &&
                errors.firstName &&
                <Alert messages={errors.firstName} />
              }
            /><br />
            <TextField
              floatingLabelText="Last Name"
              name="lastName"
              value={account.lastName}
              onChange={onTextChange}
              fullWidth
              errorText={errors && errors.lastName && <Alert messages={errors.lastName} />}
            /><br />
            <TextField
              floatingLabelText="Company Name"
              name="companyName"
              value={account.companyName}
              onChange={onTextChange}
              fullWidth
              errorText={
                errors &&
                errors.companyName &&
                <Alert messages={errors.companyName} />
              }
            /><br />
            <TextField
              floatingLabelText="Phone Number"
              name="phoneNumber"
              type="tel"
              value={account.phoneNumber}
              onChange={onTextChange}
              fullWidth
              errorText={
                errors &&
                errors.phoneNumber &&
                <Alert messages={errors.phoneNumber} />
              }
            /><br />
            <TextField
              floatingLabelText="Email"
              name="email"
              type="email"
              value={account.email}
              onChange={onTextChange}
              fullWidth
              errorText={errors && errors.email && <Alert messages={errors.email} />}
            /><br />
            <TextField
              floatingLabelText="Password"
              name="password"
              type="password"
              value={account.password}
              onChange={onTextChange}
              fullWidth
              errorText={errors && errors.password && <Alert messages={errors.password} />}
            /><br />
            <TextField
              floatingLabelText="Repeat password"
              name="passwordConfirmation"
              type="password"
              value={account.passwordConfirmation}
              onChange={onTextChange}
              fullWidth
              errorText={
                errors &&
                errors.passwordConfirmation &&
                <Alert messages={errors.passwordConfirmation} />
              }
            /><br />
          </Spacer>
          <Checkbox
            label="I agree to the terms and conditions"
            name="agreedToGeneralTerms"
            checked={account.agreedToGeneralTerms}
            onCheck={onCheckboxChange}
          />
          {
            errors &&
            errors.agreedToGeneralTerms &&
            <Alert messages={errors.agreedToGeneralTerms} />
          }
          <DialogButton type="submit" label="Sign up" primary fullWidth />
        </form>
      </DialogBody>
    </Dialog>
  )
}

LoginForm.propTypes = {
  errors: PropTypes.object.isRequired,
  onSubmit: PropTypes.func.isRequired,
  onTextChange: PropTypes.func.isRequired,
  onCheckboxChange: PropTypes.func.isRequired,
  account: PropTypes.shape({
    firstName: PropTypes.string.isRequired,
    lastName: PropTypes.string.isRequired,
    companyName: PropTypes.string.isRequired,
    phoneNumber: PropTypes.string.isRequired,
    email: PropTypes.string.isRequired,
    password: PropTypes.string.isRequired,
    passwordConfirmation: PropTypes.string.isRequired,
    agreedToGeneralTerms: PropTypes.bool.isRequired
  }).isRequired
}

export default LoginForm
