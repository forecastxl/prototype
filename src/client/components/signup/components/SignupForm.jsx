import React, { PropTypes } from 'react'
import Checkbox from 'material-ui/Checkbox'
import TextField from 'material-ui/TextField'
import { Spacer } from '../../spacer'
import { Alert } from '../../alert'

function LoginForm({ onSubmit, onTextChange, onCheckboxChange, account, errors }) {
  return (
    <div>
      <div title="Maak een account aan bij ForecastXL" titleColor="white" />
      <div>
        {errors && errors.base && <Alert messages={errors.base} />}
        <form action="/" onSubmit={onSubmit}>
          <Spacer size={'2rem'} bottom>
            <h2>Beheerder</h2>
            <p>
              Gegevens van de beheerder van dit account. Deze persoon kan bedrijven bewerken,
              toevoegen en verwijderen.
            </p>
            <TextField
              floatingLabelText="Voornaam"
              name="firstName"
              value={account.firstName}
              onChange={onTextChange}
              fullWidth
              errorText={errors && errors.firstName && <Alert messages={errors.firstName} />}
            />
            <br />
            <TextField
              floatingLabelText="Achternaam"
              name="lastName"
              value={account.lastName}
              onChange={onTextChange}
              fullWidth
              errorText={errors && errors.lastName && <Alert messages={errors.lastName} />}
            />
            <br />
            <TextField
              floatingLabelText="Email"
              name="email"
              type="email"
              value={account.email}
              onChange={onTextChange}
              fullWidth
              errorText={errors && errors.email && <Alert messages={errors.email} />}
            />
            <br />
            <TextField
              floatingLabelText="Wachtwoord"
              name="password"
              type="password"
              value={account.password}
              onChange={onTextChange}
              fullWidth
              errorText={errors && errors.password && <Alert messages={errors.password} />}
            />
            <br />
            <TextField
              floatingLabelText="Herhaal wachtwoord"
              name="passwordConfirmation"
              type="password"
              value={account.passwordConfirmation}
              onChange={onTextChange}
              fullWidth
              errorText={
                errors &&
                  errors.passwordConfirmation &&
                  <Alert messages={errors.passwordConfirmation} />
              }
            />
            <br />
          </Spacer>
          <Spacer size={'2rem'} bottom>
            <h2>Bedrijfsgegevens</h2>
            <p>
              Bedrijfsgegevens voor de facturatie. Op basis van deze gegevens maken wij voor u
              ook alvast een voorbeeld bedrijf aan.
            </p>
            <TextField
              floatingLabelText="Bedrijfsnaam"
              name="companyName"
              value={account.companyName}
              onChange={onTextChange}
              fullWidth
              errorText={errors && errors.companyName && <Alert messages={errors.companyName} />}
            />
            <br />
            <TextField
              floatingLabelText="Telefoonnummer"
              name="phoneNumber"
              type="tel"
              value={account.phoneNumber}
              onChange={onTextChange}
              fullWidth
              errorText={errors && errors.phoneNumber && <Alert messages={errors.phoneNumber} />}
            />
          </Spacer>
          <Checkbox
            label="Ik ga akkoord met de voorwaarden"
            name="agreedToGeneralTerms"
            checked={account.agreedToGeneralTerms}
            onCheck={onCheckboxChange}
          />
          {errors &&
            errors.agreedToGeneralTerms &&
            <Alert messages={errors.agreedToGeneralTerms} />}
          <button type="submit" label="Maak account aan" primary fullWidth />
        </form>
      </div>
    </div>
  )
}

LoginForm.propTypes = {
  errors: PropTypes.object.isRequired,
  onSubmit: PropTypes.func.isRequired,
  onTextChange: PropTypes.func.isRequired,
  onCheckboxChange: PropTypes.func.isRequired,
  account: PropTypes.shape({
    firstName: PropTypes.string.isRequired,
    lastName: PropTypes.string.isRequired,
    companyName: PropTypes.string.isRequired,
    phoneNumber: PropTypes.string.isRequired,
    email: PropTypes.string.isRequired,
    password: PropTypes.string.isRequired,
    passwordConfirmation: PropTypes.string.isRequired,
    agreedToGeneralTerms: PropTypes.bool.isRequired
  }).isRequired
}

export default LoginForm
