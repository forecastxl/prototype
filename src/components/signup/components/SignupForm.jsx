import React, { PropTypes } from 'react'
import styled from 'styled-components'
import Checkbox from 'material-ui/Checkbox'
import MenuItem from 'material-ui/MenuItem'
import RaisedButton from 'material-ui/RaisedButton'
import SelectField from 'material-ui/SelectField'
import TextField from 'material-ui/TextField'

const Aligner = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
`

function LoginForm({ onSubmit, onChange, account, errors }) {
  return (
    <Aligner>
      <form action="/" onSubmit={onSubmit}>
        <TextField
          floatingLabelText="First Name"
          name="firstName"
          value={account.firstName}
          onChange={onChange}
          errorText={errors.base && errors.base.join(', ')}
        /><br />
        <TextField
          floatingLabelText="Last Name"
          name="lastName"
          value={account.lastName}
          onChange={onChange}
          errorText={errors.base && errors.base.join(', ')}
        /><br />
        <TextField
          floatingLabelText="Company Name"
          name="companyName"
          value={account.companyName}
          onChange={onChange}
          errorText={errors.base && errors.base.join(', ')}
        /><br />
        <TextField
          floatingLabelText="Phone Number"
          name="phoneNumber"
          type="tel"
          value={account.phoneNumber}
          onChange={onChange}
          errorText={errors.base && errors.base.join(', ')}
        /><br />
        <SelectField
          floatingLabelText="Locale"
          name="locale"
          value={account.locale}
          onChange={onChange}
          errorText={errors.base && errors.base.join(', ')}
        >
          <MenuItem value={'nl'} primaryText="Nederlands" />
          <MenuItem value={'en'} primaryText="English" />
        </SelectField><br />
        <TextField
          floatingLabelText="Email"
          name="email"
          type="email"
          value={account.email}
          onChange={onChange}
          errorText={errors.base && errors.base.join(', ')}
        /><br />
        <TextField
          floatingLabelText="Password"
          name="password"
          type="password"
          value={account.password}
          onChange={onChange}
          errorText={errors.base && errors.base.join(', ')}
        /><br />
        <TextField
          floatingLabelText="Repeat password"
          name="passwordConfirmation"
          type="password"
          value={account.passwordConfirmation}
          onChange={onChange}
          errorText={errors.base && errors.base.join(', ')}
        /><br />
        <Checkbox
          label="I agree to the terms and conditions"
        /><br />
        <TextField
          floatingLabelText="Do you agree with our terms"
          name="agreedToGeneralTerms"
          value={account.agreedToGeneralTerms}
          onChange={onChange}
          errorText={errors.base && errors.base.join(', ')}
        /><br />
        <RaisedButton type="submit" label="Sign up" primary />
      </form>
    </Aligner>
  )
}

LoginForm.propTypes = {
  errors: PropTypes.object.isRequired,
  onSubmit: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  account: PropTypes.shape({
    firstName: PropTypes.string.isRequired,
    lastName: PropTypes.string.isRequired,
    companyName: PropTypes.string.isRequired,
    phoneNumber: PropTypes.string.isRequired,
    locale: PropTypes.string.isRequired,
    email: PropTypes.string.isRequired,
    password: PropTypes.string.isRequired,
    passwordConfirmation: PropTypes.string.isRequired,
    agreedToGeneralTerms: PropTypes.string.isRequired
  })
}

export default LoginForm
