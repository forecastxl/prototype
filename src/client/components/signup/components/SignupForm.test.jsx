import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import { DumbSignupForm } from './SignupForm'

describe('<DumbSignupForm />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<DumbSignupForm handleSubmit={() => {}} pristine submitting={false} />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
