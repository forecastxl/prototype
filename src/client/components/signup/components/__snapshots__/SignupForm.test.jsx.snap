// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DumbSignupForm /> renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <h2>
    Beheerder
  </h2>
  <p>
    Gegevens van de beheerder van dit account. Deze persoon kan bedrijven bewerken, toevoegen en verwijderen.
  </p>
  <Field
    component={[Function]}
    fullWidth={true}
    label="Voornaam"
    name="firstName"
  />
  <Field
    component={[Function]}
    fullWidth={true}
    label="Achternaam"
    name="lastName"
  />
  <Field
    component={[Function]}
    fullWidth={true}
    label="Wachtwoord"
    name="password"
    type="password"
  />
  <Field
    component={[Function]}
    fullWidth={true}
    label="Herhaal wachtwoord"
    name="passwordConfirmation"
    type="password"
  />
  <h2>
    Bedrijfsgegevens
  </h2>
  <p>
    Bedrijfsgegevens voor de facturatie. Op basis van deze gegevens maken wij voor u ook alvast een voorbeeld bedrijf aan.
  </p>
  <Field
    component={[Function]}
    fullWidth={true}
    label="Bedrijfsnaam"
    name="companyName"
  />
  <Field
    component={[Function]}
    fullWidth={true}
    label="Telefoonnummer"
    name="phoneNumber"
  />
  <Field
    component={[Function]}
    label="Ik ga akkoord met de algemene voorwaarden"
    name="agreedToGeneralTerms"
  />
  <RaisedButton
    disabled={true}
    fullWidth={true}
    label="Maak account aan"
    labelPosition="after"
    primary={true}
    secondary={false}
    type="submit"
  />
</form>
`;
